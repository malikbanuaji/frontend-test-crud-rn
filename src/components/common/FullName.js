import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {Spacing, Typography} from '../../styles';

export default function FullName({firstName, lastName}) {
  return (
    <Text key={'name'} style={styles.name}>{`${firstName} ${lastName}`}</Text>
  );
}

const styles = StyleSheet.create({
  name: {
    ...Typography.contactTitle,
    marginHorizontal: Spacing.base,
  },
});
