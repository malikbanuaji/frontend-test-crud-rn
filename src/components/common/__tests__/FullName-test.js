import React from 'react';
import renderer from 'react-test-renderer';
import FullName from '../FullName';

test('renders correctly ', () => {
  const tree = renderer
    .create(<FullName firstName={'John'} lastName={'Doe'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly without firstName', () => {
  const tree = renderer.create(<FullName lastName={'Doe'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly without lastName', () => {
  const tree = renderer.create(<FullName firstName={'John'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
