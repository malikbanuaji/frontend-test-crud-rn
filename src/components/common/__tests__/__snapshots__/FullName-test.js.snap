// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
<Text
  style={
    Object {
      "fontFamily": "Rubik-Medium",
      "fontSize": 28,
      "marginHorizontal": 16,
    }
  }
>
  John Doe
</Text>
`;

exports[`renders correctly without firstName 1`] = `
<Text
  style={
    Object {
      "fontFamily": "Rubik-Medium",
      "fontSize": 28,
      "marginHorizontal": 16,
    }
  }
>
  undefined Doe
</Text>
`;

exports[`renders correctly without lastName 1`] = `
<Text
  style={
    Object {
      "fontFamily": "Rubik-Medium",
      "fontSize": 28,
      "marginHorizontal": 16,
    }
  }
>
  John undefined
</Text>
`;
